pkg/osvscanner/optional_enricher.go

```diff
@@ -0,0 +1,43 @@
+package osvscanner
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/google/osv-scanner/pkg/lockfile"
+	"github.com/google/osv-scanner/pkg/reporter"
+)
+
+// Get the go version of the default environment that osv-scanner is being ran on
+func getGoVersion() (string, error) {
+	// GOROOT will return either the custom go location specified by $GOROOT environment
+	// variable, or it will return the default go location used during the go build,
+	// which will match the default of the platform being ran on.
+	versionBytes, err := os.ReadFile(filepath.Join(runtime.GOROOT(), "VERSION"))
```

Comment:
the GOROOT could cause a obscure bug in the future, since the default GOROOT value is the GOROOT value used to **build** the binary, not the default GOROOT for go, if we ever start building osv-scanner from a custom GO location, this will be incorrect. 

I haven't found a good alternative though, I don't think hard-coding the default GOROOT for each platform we support and switch between them is a good idea either.

Reply:
Reading from `runtime` package clearly has the disadvantages you've described.

How about relying on the environment and doing it like eg. `govulncheck` itself?
https://go.googlesource.com/vuln/+/refs/heads/master/internal/scan/run.go#84

Reply:
Good point, updated to match the `govulncheck` implementation.